### PR TITLE
fix: remove repair-malformed-updates deprecated flag

### LINKED ISSUE #490: Custom hyperkube v1.14.0-alpha.3 deployment failed
**Is this a request for help?**:
yes
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of aks-engine?**:
HEAD 08f0555
---

**Kubernetes version**:
1.14

**What happened**:
Flag `--repair-malformed-updates` was removed in https://github.com/kubernetes/kubernetes/pull/73663, which is causing the apiserver to fail to start with `unknown flag: --repair-malformed-updates`. This is causing both k8s/k8s and cloud-provider-azure prow jobs to fail.

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):
Use hyperkube image v1.14.0-alpha.3 or build from HEAD

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by ritazh**
**Reason for Change**:

Remove `repair-malformed-updates` deprecated flag

**Issue Fixed**:

Fixes #490 

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: docs/topics/clusterdefinitions.md [hunk 1]
```diff
@@ -436,7 +436,6 @@ Below is a list of apiserver options that are _not_ currently user-configurable,
 | "--etcd-keyfile"                            | "/etc/kubernetes/certs/etcdclient.key"                                                  |
 | "--etcd-servers"                            | _calculated value that represents etcd servers_                                         |
 | "--profiling"                               | "false"                                                                                 |
-| "--repair-malformed-updates"                | "false"                                                                                 |
```
Comment (tariq1890) [inline docs/topics/clusterdefinitions.md]: I understand that this is removed for k8s 1.14. Does the removal of this have consequences for other k8s clusters?

**Comment by PatrickLang** [comment]
cc @adelina-t this could be breaking Windows test passes too

**Comment by feiskyer** [comment]
Thanks for the fix.

File: docs/topics/clusterdefinitions.md [hunk 2]
```diff
@@ -436,7 +436,7 @@ Below is a list of apiserver options that are _not_ currently user-configurable,
...
+| "--repair-malformed-updates"                | "false"  (_removed in v1.14_)                                                           |
```
Comment (jackfrancis) [inline docs/topics/clusterdefinitions.md]: s/removed/deprecated

**Comment by jackfrancis** [comment]
@ritazh Add this test to `defaults-apiserver_test.go`, it should pass in your PR:

```
func TestAPIServerConfigRepairMalformedUpdates(t *testing.T) {
	// Test default
	cs := CreateMockContainerService("testcluster", "1.13.0", 3, 2, false)
	cs.setAPIServerConfig()
	a := cs.Properties.OrchestratorProfile.KubernetesConfig.APIServerConfig
	if a["--repair-malformed-updates"] != "false" {
		t.Fatalf("got unexpected default value for '--repair-malformed-updates' API server config: %s",
			a["--repair-malformed-updates"])
	}

	// Validate that 1.14.0 doesn't include --repair-malformed-updates at all
	cs = CreateMockContainerService("testcluster", "1.14.0", 3, 2, false)
	cs.setAPIServerConfig()
	a = cs.Properties.OrchestratorProfile.KubernetesConfig.APIServerConfig
	if _, ok := a["--repair-malformed-updates"]; ok {
		t.Fatalf("got a value for the deprecated '--repair-malformed-updates' API server config: %s",
			a["--repair-malformed-updates"])
	}
}
```

**Pull request merged**